Adds integration test for #3396

denibertovic opened the pull request:
The provided test package (dinamo), that is part of a custom snapshot has
a malformed cabal file. Specifically, it's setting the required cabal
version to an invalid number (9001). This test makes sure that
stack does not care about this if we do not actually use the package
in our project.

Closes #3396

denibertovic commented:
cc @snoyberg

denibertovic commented:
Bah, let me know if I should get rid of the style changes in `IntegrationSpec.hs` (those were automatic). The only change there is just uncommenting "*.gz" in the `toCopyRoot` function... I'm not sure why it was commented out in the first place.

snoyberg commented:
1. Please do roll back those style changes
2. I'm pretty sure the `base-4.10.0.0.tar.gz` file isn't necessary.
3. I think the `Main.hs` file needs to copy the `custom.index.tar.gz` file to `$HOME/.stack/indices/CustomIndex/00-index.tar.gz` before running `stack build`

denibertovic commented:
@snoyberg I've added the necessary steps and removed the files we don't need, as well as the formatting. 

One note though: We don't really need the invalid `dinamo` tarball nor do we need the `custom.index.tar.gz` file anymore but I thought it would be useful to keep them around if we ever needed to re-use in similar tests (so we don't have to create them from scratch again). What do you think?

snoyberg commented on test/integration/tests/3396-package-indices/Main.hs:
```diff
@@ -0,0 +1,13 @@
+  createDirectory (home </> ".stack" </> "indices" </> "CustomIndex")
+  copy "CustomIndex/00-index.tar" (home </> ".stack"  </> "indices" </> "CustomIndex" </> "00-index.tar")
+  copy "CustomIndex/01-index.tar" (home </> ".stack"  </> "indices" </> "CustomIndex" </> "01-index.tar")
+  copy "CustomIndex/01-index.cache" (home </> ".stack"  </> "indices" </> "CustomIndex" </> "01-index.cache")
```
Better not to copy over the `cache`, especially since the format will change over time. Also, I'd be interested in whether this works without the `00-index.tar` file as well.

denibertovic replied:
@snoyberg good point. Removing the cache file. Also, 00-index.tar is not needed so removing that as well.

snoyberg commented:
I'd probably leave off the unnecessary tarballs for now, it shouldn't be too hard to recreate something similar.

denibertovic commented:
@snoyberg Updated (removed tarballs that we don't need right now).
